spec/unit/coinbase/signer_spec.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+describe Coinbase::Signer do
+  let(:server_signer_id) { SecureRandom.uuid }
+  let(:wallets) { [SecureRandom.uuid, SecureRandom.uuid, SecureRandom.uuid] }
+  let(:model) { Coinbase::Client::ServerSigner.new('server_signer_id' => server_signer_id, 'wallets' => wallets) }
+  let(:server_signer_list_model) do
+    Coinbase::Client::ServerSignerList.new(
+      'data' => [model],
+      'total_count' => 1
+    )
+  end
+  let(:empty_server_signer_list_model) do
+    Coinbase::Client::ServerSignerList.new(
+      'data' => [],
+      'total_count' => 0
+    )
+  end
+  let(:signers_api) { double('Coinbase::Client::ServerSignersApi') }
+
+  subject(:signer) { described_class.new(model) }
+
+  before do
+    allow(Coinbase::Client::ServerSignersApi).to receive(:new).and_return(signers_api)
+  end
+
+  describe '#initialize' do
+    it 'initializes a new Signer' do
+      expect(signer).to be_a(Coinbase::Signer)
+    end
+  end
+
+  describe '#id' do
+    it 'returns the signer ID' do
+      expect(signer.id).to eq(server_signer_id)
+    end
+  end
+
+  describe '#wallets' do
+    it 'returns the wallets' do
+      expect(signer.wallets).to eq(wallets)
+    end
+  end
+
+  describe '.default' do
```

Comment:
**nit**: Class method tests should typically happen before any of the instance level tests